src/main/java/org/spongepowered/api/client/Client.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.client;
+
+import org.spongepowered.api.Engine;
+import org.spongepowered.api.UnsupportedEngineException;
+
+/**
+ * Represents a typical Minecraft Client.
+ */
+public interface Client extends Engine {
+
+    /**
+     * Throws an exception stating that the client engine is not supported.
+     *
+     * @return Nothing
+     * @throws UnsupportedEngineException The client engine is not supported
+     */
+    static <T> T unsupportedEngine() throws UnsupportedEngineException {
```

Comment:
what's the point of this? Why not throw it in Sponge?

Reply:
This was used in more than one place before, and I haven't removed it yet as I'm not sure if it will be used again right now.

Reply:
Why not just make it a dummy object provider provided method? Or the actual uses for it can be DummyObjectProvider and remapped later.